app/api/server/v1/rooms.js

```diff
@@ -150,30 +150,37 @@ API.v1.addRoute('rooms.cleanHistory', { authRequired: true }, {
 	post() {
 		const findResult = findRoomByIdOrName({ params: this.bodyParams });
 
-		if (!this.bodyParams.latest) {
+		const {
+			latest,
+			oldest,
+			inclusive = false,
+			limit,
+			excludePinned,
+			filesOnly,
+			ignoreThreads,
+			ignoreDiscussion,
+			users,
+		} = this.bodyParams;
+
+		if (!latest) {
 			return API.v1.failure('Body parameter "latest" is required.');
 		}
 
-		if (!this.bodyParams.oldest) {
+		if (!oldest) {
 			return API.v1.failure('Body parameter "oldest" is required.');
 		}
 
-		const latest = new Date(this.bodyParams.latest);
-		const oldest = new Date(this.bodyParams.oldest);
-
-		const inclusive = this.bodyParams.inclusive || false;
-
 		const count = Meteor.runAsUser(this.userId, () => Meteor.call('cleanRoomHistory', {
 			roomId: findResult._id,
-			latest,
-			oldest,
+			latest: new Date(latest),
+			oldest: new Date(oldest),
 			inclusive,
-			limit: this.bodyParams.limit,
-			excludePinned: [true, 'true', 1, '1'].includes(this.bodyParams.excludePinned),
-			filesOnly: [true, 'true', 1, '1'].includes(this.bodyParams.filesOnly),
-			ignoreThreads: [true, 'true', 1, '1'].includes(this.bodyParams.ignoreThreads),
-			ignoreDiscussion: [true, 'true', 1, '1'].includes(this.bodyParams.ignoreDiscussion),
-			fromUsers: this.bodyParams.users,
+			limit,
+			excludePinned: [true, 'true', 1, '1'].includes(excludePinned),
+			filesOnly: [true, 'true', 1, '1'].includes(filesOnly),
+			ignoreThreads: [true, 'true', 1, '1'].includes(ignoreThreads),
+			ignoreDiscussion: [true, 'true', 1, '1'].includes(ignoreDiscussion),
```

Comment:
Wondering if this could be moved into a util function, e.g. `isTruthy` or `isParamSet` for the scope of APIs. Surely would make this more readable :)

Of course, this can be done in another opportunity as we move on with this important hotfix

Reply:
yep, definitely.. this should be standardized and used across the code base :+1: